app/models/manageiq/providers/autosde/inventory/parser/storage_manager.rb

```diff
@@ -181,11 +181,35 @@ def storage_service_resource_attachments
     end
   end
 
-  # This method changes capability name field from 'abstract_capability' to 'name'
-  def parse_capabilities(capabilities, field_name)
+  def parse_ems_capabilities(capabilities)
+    return capabilities unless capabilities.is_a?(Array)
```

Comment:
Is this trying to catch when capabilities is `nil` ?  Or can capabilities be other types like a Hash?
If the former then best to just `return if capabilities.nil?`  If the later what does the format of the hash look like?

Reply:
Still have a question here

Reply:
> Is this trying to catch when capabilities is `nil` ? Or can capabilities be other types like a Hash?
> If the former then best to just `return if capabilities.nil?` If the later what does the format of the hash look like?

Hi, so the parameter `capabilities` we receive here is `collector.capability_values` which holds either the array of capabilities we received from the backend, or the already populated hash of capabilities we save in the provider.

If it is the former, we would like to parse it and populate the latter. If it's already the latter, then we would like to keep it the way it is.

Reply:
Interesting feel like there's a better way to handle this but we don't have to fix that here

Reply:
if u happen to come up with a better way, we could include it in the fix you asked for in the comment below.